columnflow/tasks/framework/mixins.py

```diff
@@ -48,13 +48,13 @@ def resolve_param_values(cls, params):
         params = super().resolve_param_values(params)
 
         # add the default calibrator when empty
-        if "config_inst" in params:
-            params["calibrator"] = cls.resolve_config_default(
-                params["config_inst"],
-                params.get("calibrator"),
-                default_str="default_calibrator",
-            )
-            params["calibrator_inst"] = cls.get_calibrator_inst(params["calibrator"], params)
+        params["calibrator"] = cls.resolve_config_default(
```

Comment:
IIRC, there were edge cases when "config_inst" was not part of params. In this case, the old behavior was that `params["calibrator"]` was not even set, and `cls.get_calibrator_inst` below wasn't invoked.

If these edge cases still exist, you would now assign `params["calibrator"] = None` and then run `cls.get_calibrator_inst` with None, whereas a sequence is expected there.

To be safe, could you do sth like

```python
if config_inst := params.get("config_inst"):
    params["calibrator"] = cls.resolve_config_default(
        ...,
        container=config_inst,
        ...
    )
    params["calibrator_inst"] = cls.get_calibrator_inst(params["calibrator"], params)
```

?

Reply:
sure, I can do that. I assume this is the case for selectors and producers aswell?

Reply:
I think so :+1: